File: src/Models/db/lyda/Playlist.ts
import {User} from "./User";
import {PlaylistLike} from "./PlaylistLike";
import {Entity} from "@targoninc/ts-search";
import {PlaylistTrack} from "./PlaylistTrack";
import {InteractionMetadata} from "../../InteractionMetadata";
import {TrackLike} from "./TrackLike";
import {Repost} from "./Repost";
import {Comment} from "./Comment";
import {Visibility} from "../../../Enums/Visibility";

export interface Playlist extends Entity {
    tracks?: PlaylistTrack[];
    user?: User;
    id: number;
    user_id: number;
    title: string;
    description: string;
    created_at: Date;
    updated_at: Date;
    visibility: Visibility;
    has_cover: boolean;
    secretcode: string;
    likes?: InteractionMetadata<TrackLike>;
    /*reposts?: InteractionMetadata<Repost>;
    comments?: InteractionMetadata<Comment>;*/
}

File: kotlin/com/example/lydaandroid/model/Playlist.kt
package com.example.lydaandroid.model

import java.util.Date

data class Playlist(
    val id: Int,
    val user_id: Int,
    val title: String,
    val description: String,
    val created_at: String,
    val updated_at: String,
    val visibility: Visibility,
    val has_cover: Boolean,
    val secretcode: String,
    val user: User? = null,
    val likes: TrackLikeMetadata? = null
)
